java/fury-core/src/main/java/org/apache/fury/Fury.java

```diff
@@ -339,12 +341,14 @@ private void write(MemoryBuffer buffer, Object obj) {
       buffer.writeInt32(-1); // preserve 4-byte for nativeObjects start offsets.
     }
     // reduce caller stack
-    if (!refResolver.writeRefOrNull(buffer, obj)) {
+    boolean isNull = refResolver.writeRefOrNull(buffer, obj);
+    if (!isNull) {
       ClassInfo classInfo = classResolver.getOrUpdateClassInfo(obj.getClass());
       classResolver.writeClass(buffer, classInfo);
       writeData(buffer, classInfo, obj);
     }
-    if (shareMeta) {
+    MetaContext metaContext = getSerializationContext().getMetaContext();
+    if (shareMeta && !isNull && metaContext != null && !metaContext.writingClassDefs.isEmpty()) {
```

Comment:
```suggestion
    if (shareMeta && !metaContext.writingClassDefs.isEmpty()) {
```

Reply:
When obj is null, can we also not execute `classResolver.writeClassDefs(buffer)`?

Reply:
When obj is null, the writingClassDefs is empty

Reply:
got, `metaContext.writingClassDefs` may cause NPE risk, maybe we should check `assert metaContext != null` before using `metaContext.writingClassDefs`

Reply:
then we can just write `metaContext != null && !metaContext.writingClassDefs.isEmpty()`

Reply:
updated